UnitTests/GitUITests/CommandsDialogs/FormCommitTests.cs

```diff
@@ -277,6 +277,59 @@ public void editFileToolStripMenuItem_Click_no_selection_should_not_throw()
             });
         }
 
+        [Test]
+        public void Dialog_remembers_window_geometry()
+        {
+            RunTwoFormsConsecutively((form1, form2) =>
+            {
+                form2.GetTestAccessor().WindowPosition
+                    .Should()
+                    .Be(form1.GetTestAccessor().WindowPosition);
+            });
+        }
+
+        [Test]
+        public void MessageBox_remembers_geometry()
+        {
+            RunTwoFormsConsecutively((form1, form2) =>
+            {
+                form2.GetTestAccessor().Message.Size
+                    .Should()
+                    .Be(form1.GetTestAccessor().Message.Size);
+            });
+        }
```

Comment:
I don't know why this test fails. On my machine it runs fine. Has someone a clue what could cause the difference between AppVeyor and my local machine?

Reply:
```
1) Failed : GitUITests.CommandsDialogs.CommitDialog.FormCommitTests.MessageBox_remembers_geometry
Expected object to be {Width=330, Height=288}, but found {Width=330, Height=292}.
   at FluentAssertions.Execution.LateBoundTestFramework.Throw(String message)
   at FluentAssertions.Execution.TestFrameworkProvider.Throw(String message)
   ....

2) Failed : GitUITests.CommandsDialogs.CommitDialog.FormCommitTests.SelectedDiff_remembers_geometry
Expected object to be {Width=513, Height=282}, but found {Width=513, Height=278}.
   at FluentAssertions.Execution.LateBoundTestFramework.Throw(String message)
   at FluentAssertions.Execution.TestFrameworkProvider.Throw(String message)
   ....
```

In both cases the different is in height, 4px.

I'll pull it down locally and check.

Reply:
Sorry, I could have included these error messages myself.

I guess, this 4 pixel gap is exactly the value of `SplitContainer.SplitterWidth`. The mentioned test result indicates that the splitter above the commit message is moving between closing and opening the commit dialog - the very thing that this PR tries to fix

Reply:
In addition, we have a race condition in `UITest.RunForm`, which will be fixed by #7394.
On AppVeyor, the tests are run by a different test runner than in MSVS -- on a single core with hyperhreading.

Reply:
@mstv I changed the code according to your feedback. Is the mentioned race condition the reason why it is currently not really possible to use the debugger with RunForm-Tests?

Reply:
Single stepping into asynchronous calls is not feasible. But you can set breakpoints inside the lambda functions.
What I mean with "race condition" is that the execution of parallel tasks/threads randomly yields wrong results caused by different timing, resulting in different sequence of executed statements.
Is there a better word?